packages/ethers/src/index.ts

```diff
@@ -133,19 +142,19 @@ export class TurnkeySigner extends ethers.Signer implements TypedDataSigner {
     return signedTx;
   }
 
-  async signTransaction(
-    transaction: ethers.utils.Deferrable<ethers.providers.TransactionRequest>
-  ): Promise<string> {
-    const unsignedTx = await ethers.utils.resolveProperties(transaction);
-
+  async signTransaction(transaction: TransactionRequest): Promise<string> {
```

Comment:
i think i found another example of someone who tried to use an external signer with ethers v6! https://github.com/ethers-io/ext-signer-ledger/blob/b59a757be7c6b0a4db823ab2903df70f8c52d0af/src.ts/signer-ledger.ts#L84

taking some heavy inspiration from their solution, but we could end up with something like:

```
import { copyRequest, resolveProperties, resolveAddress } from "ethers";

...

async signTransaction(transaction: TransactionRequest): Promise<string> {
    transaction = copyRequest(transaction);
    const { to, from } = await resolveProperties({
      to: transaction.to
        ? resolveAddress(transaction.to, this.provider)
        : undefined,
      from: transaction.from
        ? resolveAddress(transaction.from, this.provider)
        : undefined,
    });

    if (to != null) {
      transaction.to = to;
    }
    if (from != null) {
      transaction.from = from;
    }

    const tx = Transaction.from(<TransactionLike<string>>transaction);
    const unsignedTx = tx.unsignedSerialized.substring(2);

    const signedTx = await this._signTransactionWithErrorWrapping(
      unsignedTx
    );

    return `0x${signedTx}`;
  }
```

the `copyRequest` allows us to maintain all the necessary pieces of the tx! and excitingly, this means that we'd no longer have to override `sendTransaction`!

Reply:
ooo I like this a lot, _much_ cleaner!

Reply:
I kept the ens resolveAddress methods as I feel that this could be useful! I can remove if you think otherwise